resolver/src/system_conf/windows.rs

```diff
@@ -0,0 +1,55 @@
+use std::io;
+use std::str::FromStr;
+use std::net::SocketAddr;
+
+use ipconfig::get_adapters;
+use ipconfig::computer::{get_search_list, get_domain, is_round_robin_enabled};
+
+use trust_dns::rr::Name;
+use config::{ResolverConfig, ResolverOpts, NameServerConfig, Protocol};
+
+
+macro_rules! map_ipconfig_to_io_error {
+    ($result:expr) => ($result.map_err(|e| io::Error::new(io::ErrorKind::Other, Box::new(e))));
+}
+
+/// Returns the name servers of the computer (of all adapters)
+fn get_name_servers() -> io::Result<Vec<NameServerConfig>> {
+    let adapters = map_ipconfig_to_io_error!(get_adapters())?;
+    let name_servers = adapters
+        .iter()
+        .flat_map(|adapter| adapter.dns_servers().iter())
+        .map(|ipaddr| {
+                 NameServerConfig {
+                     socket_addr: SocketAddr::new(*ipaddr, 53),
+                     protocol: Protocol::Udp,
+                 }
+             })
+        .collect::<Vec<NameServerConfig>>();
+
+    Ok(name_servers)
+}
+
+pub(crate) fn read_system_conf() -> io::Result<(ResolverConfig, ResolverOpts)> {
+    let name_servers = get_name_servers()?;
+
+    let search_list: Vec<Name> = map_ipconfig_to_io_error!(get_search_list())?
+        .iter()
+        .map(|x| Name::from_str(x))
+        .collect::<Result<Vec<_>, _>>()?;
+
+    let domain = match map_ipconfig_to_io_error!(get_domain())? {
+        Some(domain) => Name::from_str(&domain)?,
+        None => Name::root(),
+    };
+
+    let config = ResolverConfig::from_parts(domain, search_list, name_servers);
+
+    let rotate = map_ipconfig_to_io_error!(is_round_robin_enabled())?;
```

Comment:
is this the only option in the Windows registry? Currently there is no logic to RoundRobin record results (which is why I have that field hidden from docs), but it's cool that this will be there when/if that gets added to the library!

Reply:
Yes I think it's the only one (that is documented). I took it from here: https://technet.microsoft.com/en-us/library/dd197418(v=ws.10).aspx

Reply:
It looks like `EnableEDNSProbes` could also be used. This would correspond to the `ResolverOpts.edns0` option. Not necessary for this change, again, I'm not doing anything explicit with edns in the Resolver at-the-moment, but plan to at some point in the future.

Reply:
It looks relevant only to the dns server that comes with windows server machines, not for the dns client service.